samza-core/src/main/java/org/apache/samza/table/caching/guava/GuavaCacheTable.java

```diff
@@ -69,36 +87,48 @@ public void put(K key, V value) {
     }
   }
 
+  @Override
+  public CompletableFuture<Void> putAsync(K key, V value) {
+    put(key, value);
```

Comment:
This should happen within the future

Reply:
same comment as deleteAsync in LocalStoreBackedReadWriteTable.